tests/h/h_api/bulk_api/entry_point_test.py

```diff
@@ -17,31 +18,22 @@ class TestBulkAPI:
     # This is a glue library, so there's not much to do here but test the
     # interfaces
 
-    def test_from_lines(self, lines, executor, CommandProcessor):
-        BulkAPI.from_lines(lines, executor)
-
-        CommandProcessor.assert_called_once_with(
-            executor=executor, observer=Any.instance_of(Observer)
-        )
-
-        self._assert_process_called_with_generator_of_commands(CommandProcessor)
-
-    def test_from_string(self, nd_json, executor, CommandProcessor):
-        BulkAPI.from_string(nd_json, executor)
-
-        CommandProcessor.assert_called_once_with(
-            executor=executor, observer=Any.instance_of(Observer)
-        )
-
-        self._assert_process_called_with_generator_of_commands(CommandProcessor)
-
-    def test_from_byte_stream(self, nd_json_byte_stream, executor, CommandProcessor):
-        BulkAPI.from_byte_stream(nd_json_byte_stream, executor)
+    @pytest.mark.parametrize(
+        "input_data,bulk_method",
+        (
+            ("lines", BulkAPI.from_lines),
+            ("nd_json", BulkAPI.from_string),
+            ("nd_json_byte_stream", BulkAPI.from_byte_stream),
+        ),
+        indirect=["input_data"],
+    )
+    def test_from_input(self, input_data, executor, bulk_method, CommandProcessor):
+        result = bulk_method(input_data, executor)
```

Comment:
Using some fanciness with pytest we can collapse these three different tests into one. I'm not sure if this is nice or horrible. I can't tell anymore.

I feel like there will be a point of diminishing returns with using every pytest trick in the book. It was all very confusing for me when I started.